### PR TITLE
refactor job manager part1

### PR DESCRIPTION
**Pull request by panda-sheep**
## What type of PR is this?
- [ ] bug
- [ ] feature
- [X] enhancement

#### Description:
refactor job manager, this pr mainly did the following parts
1. separation by space
2. Remove job_concurrency
Because only compact and flush, stats syntax supports job_concurrency, and job passes job_concurrency to task, but the corresponding task does not use job_concurrency.
Even using job_concurrency does not make much sense, because in a space, compactTask and flushTask are concurrent on the storage according to the number of engines. statsTask is concurrent according to the number of parts. FLAGS_max_concurrent_subtasks is enough

3. Fix bugs, such as https://github.com/vesoft-inc/nebula/issues/3743

## Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

## Release notes:

Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....

### CODE REVIEW

File: src/meta/processors/job/AdminJobProcessor.cpp [hunk 1]
```diff
@@ -27,45 +24,17 @@ void AdminJobProcessor::process(const cpp2::AdminJobReq& req) {
...
-        errorCode = nebula::error(jobId);
-        break;
-      }
-
-      JobDescription jobDesc(nebula::value(jobId), cmd, paras);
-      errorCode = jobMgr->addJob(jobDesc, adminClient_);
-      if (errorCode == nebula::cpp2::ErrorCode::SUCCEEDED) {
-        result.job_id_ref() = nebula::value(jobId);
-      }
+      errorCode = addJobProcess(req, result);
```
Comment (pengweisong) [inline src/meta/processors/job/AdminJobProcessor.cpp:33]: Will be 'addJobProcess' more clear and enough?

**Review by pengweisong — APPROVED** [review]
Good Job, LGTM

File: src/meta/processors/job/AdminJobProcessor.cpp [hunk 2]
```diff
@@ -142,5 +111,49 @@ void AdminJobProcessor::process(const cpp2::AdminJobReq& req) {
...
 }
 
+nebula::cpp2::ErrorCode AdminJobProcessor::addJobProcess(const cpp2::AdminJobReq& req,
+                                                         cpp2::AdminJobResult& result) {
+  auto cmd = req.get_cmd();
+  auto paras = req.get_paras();
+
+  // All jobs here are the space level, so the last parameter is spaceName.
+  if (paras.empty()) {
+    LOG(INFO) << "Parameter should be not empty";
```
Comment (SuperYoko) [inline src/meta/processors/job/AdminJobProcessor.cpp:121]: Prefer log with specific operation, this is too generic. 
If following logs metion this operation, Ignore this is okay.

File: src/meta/processors/job/AdminJobProcessor.cpp [hunk 3]
```diff
@@ -27,45 +24,17 @@ void AdminJobProcessor::process(const cpp2::AdminJobReq& req) {
   }
   LOG(INFO) << __func__ << "() " << oss.str();
 
-  JobManager* jobMgr = JobManager::getInstance();
+  cpp2::AdminJobResult result;
+  auto errorCode = nebula::cpp2::ErrorCode::SUCCEEDED;
```
Comment (darionyaphet) [inline src/meta/processors/job/AdminJobProcessor.cpp:28]: using `nebula::cpp2::ErrorCode` to replace `auto` would be better?

**Review by critical27 — APPROVED** [review]
Generally LGTM.

File: src/meta/processors/job/AdminJobProcessor.cpp [hunk 4]
```diff
@@ -142,5 +111,49 @@ void AdminJobProcessor::process(const cpp2::AdminJobReq& req) {
...
+  }
+
+  folly::SharedMutex::WriteHolder holder(LockUtils::lock());
+  auto jobId = autoIncrementId();
+  if (!nebula::ok(jobId)) {
+    return nebula::error(jobId);
+  }
+
+  JobDescription jobDesc(nebula::value(jobId), cmd, paras);
+  auto errorCode = jobMgr_->addJob(jobDesc, adminClient_);
```
Comment (critical27) [inline src/meta/processors/job/AdminJobProcessor.cpp:151]: The job manager will have one queue for one space, right?

Comment (panda-sheep) [reply src/meta/processors/job/AdminJobProcessor.cpp:151]: Yes, the later pr will do the following work:
Each space has a priority queue.
Jobs in the same space are executed serially according to priority.
Jobs in different spaces are executed in parallel according to the space.

File: src/meta/processors/job/AdminJobProcessor.cpp (hunk 1 above)
Comment (panda-sheep) [reply src/meta/processors/job/AdminJobProcessor.cpp:33]: My original intention is: for different job operations, separate out a separate interface.
So `addJobProcess` handles the operation of `add job`.
Of course, within the add job, there is its own check.

File: src/meta/processors/job/AdminJobProcessor.cpp (hunk 2 above)
Comment (panda-sheep) [reply src/meta/processors/job/AdminJobProcessor.cpp:121]: Good point.
The following pr will move the last `spaceName` of parameter out.
Different `AdminJobOp` operations, check the validity of their own parameters.
Therefore, the following pr will uniformly change this error message to make the error message more clear.

**Review by liuyu85cn — APPROVED** [review]
good job

**Review by CPWstatic — APPROVED** [review]

**Pull request merged**